csharp/src/Apache.Arrow/DecimalUtility.cs

```diff
@@ -52,7 +52,11 @@ internal static decimal GetDecimal(in ArrowBuffer valueBuffer, int index, int sc
                 BigInteger integerPart = BigInteger.DivRem(integerValue, scaleBy, out BigInteger fractionalPart);
                 if (integerPart > _maxDecimal || integerPart < _minDecimal) // decimal overflow, not much we can do here - C# needs a BigDecimal
                 {
-                    throw new OverflowException($"Value: {integerPart} too big or too small to be represented as a decimal");
+                    throw new OverflowException($"Value: {integerValue} is too big or too small to be represented as a decimal");
+                }
+                if (fractionalPart > _maxDecimal || fractionalPart < _minDecimal) // decimal overflow, not much we can do here - C# needs a BigDecimal
+                {
+                    throw new OverflowException($"Value: {integerValue} is too big or too small to be represented as a decimal");
```

Comment:
The test cases confused me, but looking at it, it's not necessarily about too big or too small, but rather too precise? 

Arrow represents 7.89 in `decimal256(76,38)` as `789000000000000000000000000000000000000`
integerPart is then `7`
fractionalPart is `89000000000000000000000000000000000000`
and that's just too many digits of precision for `decimal`?

Which, it is literally too big, but that's a bit confusing unless you are thinking about the representation.

Reply:
Separately, I think it would be useful to differentiate the error messages here and possibly include the problematic value

Reply:
updated in latest push